src/local-history/local-history-manager.ts

```diff
@@ -170,11 +169,37 @@ export class LocalHistoryManager {
     /**
      * Revert the current active editor to its previous revision.
      */
-    public revertActiveEditorToPrevRevision(previous: vscode.TextEditor, current: vscode.TextEditor): void {
+    public async revertActiveEditorToPrevRevision(previous: vscode.TextEditor, current: vscode.TextEditor): Promise<void> {
         if (current && previous) {
-            const fileBuffer = fs.readFileSync(previous.document.fileName);
-            fs.writeFileSync(current.document.fileName, fileBuffer);
+            const fileContent = await this.readFile(previous.document.fileName);
+            await this.writeFile(current.document.fileName, fileContent);
         }
     }
 
+    /**
+     * Reads the contents of the file.
+     * @param uri: the uri of the file to be read.
+     * @param encoding: the required format of the data received by the stream.
```

Comment:
[minor] not sure that 'format' is appropriate, but rather we should mention the 'conversion'